rotkehlchen/chain/polygon_pos/modules/polygon_pos_bridge/decoder.py

```diff
@@ -0,0 +1,166 @@
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any, Final
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.polygon.constants import CPT_POLYGON, CPT_POLYGON_DETAILS
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+WITHDRAW_TOPIC: Final = b'\xec:\xfb\x06{\xce3\xc5\xa2\x94G\x0e\xc5\xb2\x9egY0\x1c\xd3\x92\x85PI\x0cmH\x81l\xdc/]'  # noqa: E501
+DEPOSIT_FOUR_BYTES: Final = b'\x2e\x1a\x7d\x4d'
+DEPOSIT_POL_TOPIC: Final = b'\xeb\xff&\x02\xb3\xf4h%\x9e\x1e\x99\xf6\x13\xfe\xd6i\x1f:e&\xef\xfen\xf3\xe7h\xbaz\xe7\xa3lO'  # noqa: E501
+STATE_RECIEVER_ADDRESS: Final = string_to_evm_address('0x0000000000000000000000000000000000001001')
+STATE_COMMITTED_TOPIC: Final = b'Z"rU\x90\xb0\xa5\x1c\x929@"?tXQ!d\xb1\x113Y\xa75\xe8n\x7f\'\xf4G\x91\xee'  # noqa: E501
+PLASMA_BRIDGE_CHILD_CHAIN: Final = string_to_evm_address('0xD9c7C4ED4B66858301D0cb28Cc88bf655Fe34861')  # noqa: E501
+
+
+class PolygonPosBridgeDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+
+    def _decode_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes bridge withdraw events.
+
+        Ethereum to Polygon bridging uses a state sync mechanism to transfer data from ethereum
+        to polygon. See https://docs.polygon.technology/pos/architecture/bor/state-sync/
+
+        Mapped to STATE_RECIEVER_ADDRESS. When it receives a STATE_COMMITTED_TOPIC event,
+        a bridge withdrawal has been initiated and the corresponding receive event needs modified.
+        """
+        if context.tx_log.topics[0] != STATE_COMMITTED_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
Is this the logic that decodes this test that you have? https://polygonscan.com/tx/0x962b5ec3701a4441ceca761c77879ae112e5052d53e80b3a2948fc52abdf8aee#eventlog

There is a deposit event that you can use to extract the amount bridged. The filter that you do is wide and maybe can be narrowed
![imagen](https://github.com/user-attachments/assets/1dab8299-35b7-41da-85ed-06d458f4a8f3)

Reply:
No, this logic decodes the test for https://polygonscan.com/tx/0x74b36dd98fb9e3a6bf9fa4cd6ba2066cd3fa86797b206bd709cfb097bb0c85a4#eventlog where there are only state committed and transfer events. I don't think there's any good way to narrow it down further.

The logic for the transaction you linked is in `_decode_plasma_withdraw`, where it does check for the topic of the `TokenDeposited` event. This was misnamed as `WITHDRAW_TOPIC` (now changed to `TOKEN_DEPOSITED_TOPIC`). These names are slightly confusing since the name of the method we're checking for is something to do with deposit while the event type were going to be setting is actually withdraw, but it probably is best to have it named the same as the method its actually refers to.